### PR TITLE
fix: rewrite encryption system information flow

### PR DESCRIPTION
**Pull request by Unix4ever**
Pass getter to the key handler instead of already fetched node uuid.

### CODE REVIEW

File: internal/pkg/mount/system.go [hunk 1]
```diff
@@ -190,8 +190,12 @@ func SystemPartitionMount(ctx context.Context, r runtime.Runtime, logger *log.Lo
 	}
 
 	systemInformation, err := r.GetSystemInformation(ctx)
-	if err != nil && !errors.Is(err, context.Canceled) && state.IsNotFoundError(err) {
-		return err
+	if err != nil {
```
Comment (smira) [inline internal/pkg/mount/system.go:193]: I don't think this has anything to do with timeouts, as it fails after installing Talos, which is way late in the boot sequence

Comment (Unix4ever) [reply internal/pkg/mount/system.go]: yes, I found the actual bug.

**Review by smira — APPROVED** [review]

**Pull request merged**